oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/OAuth2AccessToken.java

```diff
@@ -0,0 +1,407 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.auth.oauth2;
+
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+import static java.util.Objects.requireNonNull;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+
+/**
+ * Defines a structure of the Access Token Response, as per
+ * <a href="https://tools.ietf.org/html/rfc6749#section-5.1">[RFC6749], Section 5.1</a>.
+ */
+public class OAuth2AccessToken implements Serializable {
```

Comment:
I realized that we already have a class named `OAuth2Token` at `com.linecorp.armeria.common.auth`. How about `OAuth2AccessTokenResponse`?

Reply:
I don't like the name `Response`. It implies something ephemeral. The name should imply that this object _contains_ the Access Token.

Reply:
What about `OAuth2AccessTokenRecord` or, as per you other suggestion, `GrantedOAuth2AccessToken`?

Reply:
`GrantedOAuth2AccessToken` or `GrantedOAuth2Token`? I wonder if we can drop `Access` here since a token usually means an access token. I'm not an expert on this topic, so it's your call.

Reply:
No, we should not drop `Access` for two reasons:
1. This object represents exactly the issued **Access** Token (as per [RFC6749, Section 5.1](https://tools.ietf.org/html/rfc6749#section-5.1)).
2. There are common `Token` operations that apply to different type of tokens, like `Access Token` or `Refresh Token`. For instance, `OAuth2TokenDescriptor` or `TokenRevocationRequest` may apply to either `Access` or `Refresh` tokens.

I'll go ahead and rename it to `GrantedOAuth2AccessToken` then.